### DISCUSSION TITLE
Welcome to oqtane.docs Discussions!

### DISCUSSION THREAD

**Discussion by iJungleboy** [oqtane/oqtane.docs#21] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by thabaum** [comment]
#  👍 Awesome! 👍 
Thank you for this enhancement @iJungleboy !

## About Myself
I am one of the developer contributors for the Oqtane Framework helping as I can.  I have been assisting with Oqtane through the years since beta version about 4 years ago.  I have code contributions along with a lot of general feedback I try to present to help come up with the best solution for our project lead @sbwalker to pursue.  I have worked with DNN community since about 2004/2005 and am excited to be a part of this Oqtane community 20 years later!  

DNN and Oqtane are what I have used and continue use to grow my abilities as a .NET full stack developer now for mobile and web.  Personally I liked the schema and overall architecture of DotNetNuke which is why I chose it over Wordpress, Drupal, Orchard, Umbracco and other open sourced projects.  As soon as Shaun Walker created Blazor solution with .NET Core, and DNN was staying put for eternity with .NET Framework my choice was to work with the Oqtane Framework to investigate if it would actually be a fit for my future web and mobile application plans.  

Since Oqtane Framework is a component based modular system it makes it very easy to enhance with a working out of the box solution.  As of now I plan to migrate those complex sites from DNN to the Oqtane Framework.  When I say now, I mean now I feel Oqtane is ready for me at version 5, and I am ready for Oqtane.  I also feel more comfortable now taking on this task of migrating some fairly large user sites with features I am building in custom modules now.  I feel while I am taking these steps I can be also producing some documentation ideas to share and have as an additional reference to my own notes for those working with module and .NET Maui app development.

## My Support Oqtane Documentation Community Plan
 I plan to kick down some ideas and engage with some other community members full throttle.  I want to help build momentum in this arena leading by example helping show others how they can too contribute.  With this new feedback and support keeping the documentation project fresh, what I will be doing is as I run into different areas I am contributing for in Oqtane Framework itself, I will review it's documentation and be sure it is current and full of detailed information as needed the best I can with all new features or behaviors covered.  

## The End Game
My goal here is to have a great reference for myself and the community along with freeing up as much time for our project lead to focus on more concerning issues with the framework.  I am hoping to help our documentation project lead as well coordinate and review things so this project as well can move forward in the fast lane.  I would like to see the activity bump up on my personal profile in GitHub for this project.  I want to put a 2-1 effort into docs to catch up here.  

I want to eventually just need to spend a set few hours a month on them.  Less issues with setup and so on, and extreme confidence in one source of documentation we may be able to include links in the framework to add a reliable support option.  Although this can create unwanted overhead, big problem areas may make it worth it such as installation area.  

I am hoping to motivate and encourage some of the other Oqtane Framework contributors to see the value in learning DocFX and making support information available as they make improvements or are working in an area.  At least posted in an issue for those volunteering to maintain the docs to contribute.  If we can streamline things a little and make it appealing for different levels of contributions activity should increase here.  
 
Any module developers, especially those with a commercial marketplace product should see this documentation project as an important piece of their success, our success.  I think some communication at developer meetings a quick plug in this direction may help ignite activity here.  Oqtane 5 is getting very mature now so dedicating an half-hour a month towards documentation reviews maybe worth it for everyone involved.

## What Else
I have a few contributions I have been thinking of putting up open source as well that may help once I get going here that should help people get started using the framework.  I also may eventually produce a few products for the marketplace although this is not on my current roadmap.

## Final Thoughts
Helping support documentation is a great way to get to know the framework and a great place to help support our open source community.  Oqtane Documentation I believe is great place to learn the process of submitting a pull request without too much pressure for those new to GitHub.  

As a past contributor to open source it is exciting to see them be merged.  It can also be discouraging to not see one get merged.  My advice is don't give up, comment more on that one PR if you feel strongly about it being the solution or needing something like it.  Maybe things were not delivered clearly or maybe you should just let it go and move on to the next issue you may have.  

I would love to hear about issues with the Oqtane Framework not aligning with our DocFX documentation.  Even if it is letting us all know this section needs updated to help us zero in on the places being used the most.  If you don't know how to, or just don't have time to submit a pull request by submitting an issue one of us will get on top of it eventually.

This is a 100% Oqtane Community supported project no one gets paid but we get a resource which is a value of it's own.  It's only as valuable as we make it.  I will do what I can to help create content here, make assignments with tags "Needs Help" so contributors looking for ways to help can tackle these.

Thanks again @iJungleboy for everything done in the past and future to help us have something to fall back on.  I can't wait to see the DB Schema and other enhancements made.  Now that we have a discussion board I will maybe move an issue or two here refined for discussions.

Also thanks to everyone else who have been attempting to get involved and all those who helped in the past!

Cheers!

**Reply by GregStevenson** [reply]
I would also like to thank @iJungleboy  for starting this discussion section. 

### About Myself
I've been coding minis and [micros ](https://en.wikipedia.org/wiki/KIM-1)since the 70s.  I've worked on everything from embedded medical devices to websites for the aerospace industry.  However, I just started learning Blazor, MAUI, and Oqtane this past summer. I'm currently working on the MVP for a startup and Oqtane is the platform I've chosen to build it on. If successful, I'm hoping this startup will be a  shining example of what can be achieved with Oqtane.

### Community Plans

 I have always believed that the best way to master a subject is to teach it. That is the reason I created an[ Oqtane meetup](https://www.meetup.com/socal-oqtane-meetup/) last year and have been in discussions with several members of the Oqtane community about documenting Oqtane. I figured it was the area where I could be the biggest help and do the least damage. (Your program won't crash if the documentation has a semi-colon in the wrong spot.) I don't have a formal day job and have already been documenting as I learn Oqtane on Gitbook since it uses markdown. I have been investigating some workflows that might be useful for web, print, and video approaches to documenting Oqtane. I'll try and present these next week.

**Comment by iJungleboy** [comment]
Thanks @thabaum & @GregStevenson  😉

Short intro of myself: 

* Missionary kid, grew up in Indonesia till ca. 17 - taught myself programming on the bible translation PC in hours away from civilization
* Came "back" to Switzerland in 1994, learned German and founded [2sic](https://www.2sic.com/) during junior college - a good 25 years ago
* Entered the DNN scene ca. 2003
* Because of module-hell we created a generic data-model + template system called [2sxc](https://2sxc.org/) in 2012
* It has since matured into a full fledged CMS plugin for DNN, which according to estimates of the core DNN team currently runs on about 50% of all DNN installations
* I've been a strong advocate for moving DNN to .net core since 2016 - so when Oqtane came around it was a perfect fit
* We then improved 2sxc to run on _both_ Dnn and Oqtane, giving Oqtane extensive CMS features as well as ca. 30 pre-built Apps like image galleries and more just-like-that
* I had tried many things to document 2sxc and wanted to pass on that experience to Oqtane, which is why I spearheaded the docs
